core/services/keeper/validate.go

```diff
@@ -2,65 +2,58 @@ package keeper
-                          from="[$(jobSpec.fromAddress)]"
-                          evmChainID="$(jobSpec.evmChainID)"
-                          data="$(encode_perform_upkeep_tx)"
-                          gasLimit="$(jobSpec.performUpkeepGasLimit)"
-                          txMeta="{\"jobID\":$(jobSpec.jobID),\"upkeepID\":$(jobSpec.prettyID)}"]
-encode_check_upkeep_tx -> check_upkeep_tx -> decode_check_upkeep_tx -> encode_perform_upkeep_tx -> perform_upkeep_tx`
-)
-
-// expectedPipeline it is basically parsed ExpectedObservationSource value
-var expectedPipeline pipeline.Pipeline
+const ObservationSource = `
+    encode_check_upkeep_tx   [type=ethabiencode
+                              abi="checkUpkeep(uint256 id, address from)"
+                              data="{\"id\":$(jobSpec.upkeepID),\"from\":$(jobSpec.fromAddress)}"]
+    check_upkeep_tx          [type=ethcall
+                              failEarly=true
+                              extractRevertReason=true
+                              evmChainID="$(jobSpec.evmChainID)"
+                              contract="$(jobSpec.contractAddress)"
+                              gas="$(jobSpec.checkUpkeepGasLimit)"
+                              gasPrice="$(jobSpec.gasPrice)"
+                              gasTipCap="$(jobSpec.gasTipCap)"
+                              gasFeeCap="$(jobSpec.gasFeeCap)"
+                              data="$(encode_check_upkeep_tx)"]
+    decode_check_upkeep_tx   [type=ethabidecode
+                              abi="bytes memory performData, uint256 maxLinkPayment, uint256 gasLimit, uint256 adjustedGasWei, uint256 linkEth"]
+    encode_perform_upkeep_tx [type=ethabiencode
+                              abi="performUpkeep(uint256 id, bytes calldata performData)"
+                              data="{\"id\": $(jobSpec.upkeepID),\"performData\":$(decode_check_upkeep_tx.performData)}"]
+    perform_upkeep_tx        [type=ethtx
+                              minConfirmations=0
+                              to="$(jobSpec.contractAddress)"
+                              from="[$(jobSpec.fromAddress)]"
+                              evmChainID="$(jobSpec.evmChainID)"
+                              data="$(encode_perform_upkeep_tx)"
+                              gasLimit="$(jobSpec.performUpkeepGasLimit)"
+                              txMeta="{\"jobID\":$(jobSpec.jobID),\"upkeepID\":$(jobSpec.prettyID)}"]
+    encode_check_upkeep_tx -> check_upkeep_tx -> decode_check_upkeep_tx -> encode_perform_upkeep_tx -> perform_upkeep_tx
+`
 
+// We parse the ObservationSource only once here, because it is constant for all the Keeper jobs.
 func init() {
-	pp, err := pipeline.Parse(ExpectedObservationSource)
+	_, err := pipeline.Parse(ObservationSource)
 	if err != nil {
-		panic(fmt.Sprintf("failed to parse default observation source: %v", err))
+		panic(fmt.Sprintf("Failed to parse default Keeper observation source: %v", err))
 	}
```

Comment:
I am not sure what the point of this is if the parsed value is not used anywhere